File: packages/flutter_shared/lib/tools/text_case_tool.dart
/// 文字格式互转 — 大小写 / 命名法
class TextCaseTool {
  TextCaseTool._();

  /// 所有转换结果
  static ({String upper, String lower, String title, String sentence,
      String camel, String pascal, String snake, String kebab, String reverse})
  convert(String input) {
    if (input.isEmpty) {
      return (
        upper: '', lower: '', title: '', sentence: '',
        camel: '', pascal: '', snake: '', kebab: '', reverse: '',
      );
    }

    return (
      upper: input.toUpperCase(),
      lower: input.toLowerCase(),
      title: _toTitleCase(input),
      sentence: _toSentenceCase(input),
      camel: _toCamelCase(input),
      pascal: _toPascalCase(input),
      snake: _toSnakeCase(input),
      kebab: _toKebabCase(input),
      reverse: input.split('').reversed.join(),
    );
  }

  static String _toTitleCase(String input) {
    return input.split(RegExp(r'[\s_\-]+')).map((word) {
      if (word.isEmpty) return word;
      return word[0].toUpperCase() + word.substring(1).toLowerCase();
    }).join(' ');
  }

  static String _toSentenceCase(String input) {
    if (input.isEmpty) return input;
    return input[0].toUpperCase() + input.substring(1).toLowerCase();
  }

  static List<String> _splitWords(String input) {
    // 处理 camelCase / PascalCase
    final withSpaces = input.replaceAllMapped(
      RegExp(r'([a-z])([A-Z])'),
      (m) => '${m[1]} ${m[2]}',
    );
    // 处理数字边界
    final withSpaces2 = withSpaces.replaceAllMapped(
      RegExp(r'([a-zA-Z])(\d)'),
      (m) => '${m[1]} ${m[2]}',
    );
    return withSpaces2
        .split(RegExp(r'[\s_\-]+'))
        .where((w) => w.isNotEmpty)
        .toList();
  }

  static String _toCamelCase(String input) {
    final words = _splitWords(input);
    if (words.isEmpty) return '';
    return words[0].toLowerCase() +
        words.skip(1).map((w) => _capitalize(w.toLowerCase())).join();
  }

  static String _toPascalCase(String input) {
    return _splitWords(input)
        .map((w) => _capitalize(w.toLowerCase()))
        .join();
  }

  static String _toSnakeCase(String input) {
    return _splitWords(input)
        .map((w) => w.toLowerCase())
        .join('_');
  }

  static String _toKebabCase(String input) {
    return _splitWords(input)
        .map((w) => w.toLowerCase())
        .join('-');
  }

  static String _capitalize(String s) {
    if (s.isEmpty) return s;
    return s[0].toUpperCase() + s.substring(1);
  }
}

File: miniapp/src/utils/tools/textCaseTool.ts
/**
 * 文字格式互转 — 大小写 / 命名法
 * 移植自 flutter_shared/tools/text_case_tool.dart
 */

export interface TextCaseResult {
  upper: string;
  lower: string;
  title: string;
  sentence: string;
  camel: string;
  pascal: string;
  snake: string;
  kebab: string;
  reverse: string;
}

function splitWords(input: string): string[] {
  const withSpaces = input.replace(/([a-z])([A-Z])/g, '$1 $2');
  const withSpaces2 = withSpaces.replace(/([a-zA-Z])(\d)/g, '$1 $2');
  return withSpaces2.split(/[\s_-]+/).filter(w => w.length > 0);
}

function capitalize(s: string): string {
  if (!s) return s;
  return s[0].toUpperCase() + s.substring(1).toLowerCase();
}

export const TextCaseTool = {
  convert(input: string): TextCaseResult {
    if (!input) {
      return {
        upper: '', lower: '', title: '', sentence: '',
        camel: '', pascal: '', snake: '', kebab: '', reverse: '',
      };
    }

    const words = splitWords(input);

    return {
      upper: input.toUpperCase(),
      lower: input.toLowerCase(),
      title: words.map(w => capitalize(w)).join(' '),
      sentence: input[0].toUpperCase() + input.substring(1).toLowerCase(),
      camel: words.length === 0 ? '' :
        words[0].toLowerCase() + words.slice(1).map(w => capitalize(w)).join(''),
      pascal: words.map(w => capitalize(w)).join(''),
      snake: words.map(w => w.toLowerCase()).join('_'),
      kebab: words.map(w => w.toLowerCase()).join('-'),
      reverse: input.split('').reverse().join(''),
    };
  },
};
